Reduce build warnings

jrajahalme opened the pull request:
When Cilium repo is located on a directory whose path is not a legal
Go path, go list adds a leading underscore to the output absolute
paths. Remove the leading underscore and the absolute path. This will
not make all make targets to to work (e.g. 'make gofmt' will not work
in this scenario), but a lot of spurious warnings for targets that do
not need Go are eliminated.

Filter out the annoying "warning: directory does not exist" warnings
when building protobufs. A preceding "INFO: From ProtoCompile" lines
still remain, but there are only a handful of them, and they provide
an indication that something was filtered out.

Fixes: #4703

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5031)
<!-- Reviewable:end -->

joestringer commented on envoy/Makefile:
```diff
@@ -25,6 +25,7 @@ ENVOY_BINS = \
 CHECK_FORMAT ?= ./bazel-bin/check_format.py.runfiles/envoy/tools/check_format.py
 
 BAZEL ?= $(QUIET) bazel
+BAZEL_FILTER ?= 2>&1 | grep -v -e "bazel-out/.*/genfiles/external/.*: warning: directory does not exist."
```
won't this hide the result of bazel running, and instead report the success or failure of the `grep`?

rlenglet replied:
You're right.

joestringer reviewed (dismissed):
I think you'll want to do something like this to avoid the issue I raised earlier:
```
bash -c '( set -o pipefail; <bazel build> | <filter>)'
```

This causes a failure in the first command to fail the whole line, rather than piping the output of the first (failed) command into the second command.

But then you probably need to prepend such lines with `$(QUIET)` and probably disable the `$(BAZEL)` quiet definition in `Makefile.quiet` (as `$(BAZEL)` would only be used from the bash, rather than from the Makefile). Then just need to check the right syntax to fetch the Makefile variables and use them inside this nested bash session from the command I propose above.

Maybe there's a simpler way, but I think that the above will work.

jrajahalme committed:
envoy: Filter out unhelpful warnings

Filter out the annoying "warning: directory does not exist" warnings
when building protobufs. A preceding "INFO: From ProtoCompile" lines
still remain, but there are only a handful of them, and they provide
an indication that something was filtered out.

Fixes: #4703
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
Makefile: Fix go list output when not it a valid Go path.

When Cilium repo is located on a directory whose path is not a legal
Go path, go list adds a leading underscore to the output absolute
paths. Remove the leading underscore and the absolute path. This will
not make all make targets to to work (e.g. 'make gofmt' will not work
in this scenario), but a lot of spurious warnings for targets that do
not need Go are eliminated.

Fixes: #4703
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
